accepted/install-locations.md

```diff
@@ -0,0 +1,227 @@
+
+*This type of install is supported by .NET Core 2.2 and earlier.*
+
+### Self-contained apps
+Technically speaking self-contained apps also "install" a private copy of the runtime, but they don't use the same layout and such installation is only usable by the app itself. So for the purposes of this document, self-contained apps are largely out of scope.
+
+
+## Components
+These are the components of .NET Core which must be aware of the install location:
+* The host (muxer (`dotnet`), `apphost`, `comhost`, ...) - the host needs to be able to find the location to search for
+  * Latest version of `hostfxr` to use.
+  * Shared frameworks
+  * SDKs
+* Installer - the installer needs to put the files into the right layout and register the install location with the system configuration (if applicable)
+* Applications
+  * If the application is using the muxer (`dotnet`) it needs to be able to find it. This applies to tools like MSBuild or VS as well.
+  * Certain tools (installers, development tools) may need to determine if .NET Core is installed and which versions are available.
+
+
+## Servicing considerations
+The algorithm which determines where to search, is to some degree contained in all the components mentioned above. Changes to the algorithm (for example introduction of a new default location or new install type) need to be applied to all of these components.
+
+Unfortunately some of these components are hard to service:
+* The muxer relies on the caller to locate it, then it's shared by all versions installed in any given location. Updating it is troublesome as it typically require a system reboot (may be in use by running applications).
+* The `hostfxr` is relatively easy to service since the installer places a newer version side-by-side and relies on the host to pick the latest available version. The complexity comes from the fact that it has to work with all versions of the runtime/sdk.
+* App-contained hosts (`apphost`, `comhost`, `ijwhost`, `nethost`) these are libraries which are shipped with the apps and thus fully owned/serviced by the app. They need to know how to find install locations to locate shared frameworks.
+
+The biggest problem is with the app-contained hosts which we can't service in any way. So introducing new install types or changing the default location can be breaking for apps using old hosts.
+
+The problem already exists since the `apphost` has been shipped in .NET Core 2.1 with support for framework dependent apps. Fortunately this feature is not widely used yet.
+
+In .NET Core 3.0 the situation will be very different. Framework dependent `apphost` is the default for apps. Scenarios requiring the other hosts are also likely to be relatively common (COM, IJW, native hosting). So the hosts which we ship with .NET Core 3.0 will be spread across all machines running .NET Core (not just development but production as well) without any ability to service them directly.
+
```

Comment:
(minus) In .NET Core 3.0 the situation will be very different. Framework dependent `apphost` is the default for apps.
(add) In .NET Core 3.0 framework dependent apps use an `apphost` by default, which makes this case far more prevalent.